src/utilities/CMakeLists.txt

```diff
@@ -67,8 +80,19 @@ else()
         "Should be \"TBB\" or \"NONE\"")
 endif()
 
-target_include_directories(${PROJECT_NAME} INTERFACE ${THRUST_INC_DIR})
-target_link_libraries(${PROJECT_NAME} INTERFACE glm)
+target_include_directories(${PROJECT_NAME} INTERFACE ${PROJECT_SOURCE_DIR}/include)
+target_link_libraries(${PROJECT_NAME} INTERFACE glm::glm)
+if(DEFINED Thrust_SOURCE_DIR)
+target_include_directories(${PROJECT_NAME} INTERFACE
+    ${Thrust_SOURCE_DIR}
+    ${Thrust_SOURCE_DIR}/dependencies/libcudacxx/include
+)
+else()
+target_include_directories(${PROJECT_NAME} INTERFACE
+    ${THRUST_ROOT}
+    ${THRUST_ROOT}/dependencies/libcudacxx/include
```

Comment:
I'm a little surprised CMake doesn't take care of this automatically...

Reply:
CMake should be able to handle it, but I found that for some reason FetchContent does not execute the thrust cmake file correctly. This is what we did previously as well when using submodules.